### PR TITLE
Cart btn

### PR DESCRIPTION
**Pull request by MarikaBBB**
Refactor ShopCart.tsx: Extract TotalPrice and Quantifier Components and Fix Integration Issues: 
- The TotalPrice and Quantifier component - Display logic into their own component.
- Adjusted imports to include the new TotalPrice and Quantifier components.
- Replaced inline code with the new components to streamline ShopCart.tsx.
- Corrected the prop name from onIcrease to onIncrease in the Quantifier component and its usages.
- Fixed the misspelling of Quantifier (was Quentifier) throughout the codebase.
- Fixed parameter mismatches in the updateCartState function.
Screenshot 2024-10-23 at 14 47 17

Add style and refined it.

### CODE REVIEW

**Comment by MarikaBBB** [comment]
Final style

Screenshot 2024-10-23 at 16 56 14

File: cypress/e2e/log-in.cy.ts [hunk 1]
```diff
@@ -26,8 +27,8 @@ describe('User Authentication Tests', () => {
     cy.get('button[type="submit"]').click();
 
     // error message after login atempt
-    cy.get('[class=.error-message]').should('be.visible')
-      .and('contain', 'Email or password is incorrect');
+    cy.get('.error-message').should('be.visible').and('contain', 'Email or password is incorrect');
```
Comment (maxitect) [inline cypress/e2e/log-in.cy.ts]:
```suggestion
    cy.get('span[class="error-message"]').should('be.visible').and('contain', 'Email or password is incorrect');
```

File: src/Cart-components/Quantifier.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            className="rounded text-white font-bold bg-background-footer text-white hover:bg-accent-light transition w-12 h-12 flex justify-center items-center" 
+            onClick={onDecrease}>
+                -
+            </button>
+            <span className="rounded text-lg bg-white w-12 h-12 flex justify-center items-center">{quantity}</span>
+            <button 
+            className="rounded text-white font-bold bg-background-footer text-white hover:bg-accent-light transition w-12 h-12 flex justify-center items-center"  
+            onClick={onIncrease}>
+                +
+            </button>
```
Comment (maxitect) [inline src/Cart-components/Quantifier.tsx]: Should these buttons be one component that can be re-used for + and - using props?

File: src/Cart-components/TotalPrice.tsx
Comment (maxitect) [inline src/Cart-components/TotalPrice.tsx:1]: Could this be a component used both for total price and for individual line prices (per vinyl * quantity) using props?

File: src/components/layout/Footer.tsx [hunk 3]
```diff
@@ -2,11 +2,10 @@ const Footer = () => {
     return (
       <footer
       role="contentinfo"
-      className="bg-background-footer text-text-primary py-8 w-full"
-    >
-        <p className="text-center">&copy; 2024 Font Hill Records. All rights reserved.</p>  
-      
-      </footer>
+      className="bg-background-footer text-text-primary py-8 w-full">
+        <p className="text-center">© 2024 Font Hill Records. All rights reserved.</p>
```
Comment (maxitect) [inline src/components/layout/Footer.tsx]:
```suggestion
        <p className="text-center">&copy; 2024 Font Hill Records. All rights reserved.</p>
```
We should keep using normal html notation for symbols, why the change?

File: src/pages/ShopCart.tsx [hunk 4]
```diff
@@ -1,95 +1,112 @@
...
-        );
-        setCartCount(newCartCount);
-      };
-    
+  return (
+    <section 
+    id="cart"
+    className="bg-background-light p-4 shadow rounded-t-lg"
+    >
+      <h1 className="text-4xl font-bold mb-4 text-text-primary text-center">Shopping Cart</h1>
```
Comment (maxitect) [inline src/pages/ShopCart.tsx]:
```suggestion
      <h2 className="text-4xl font-bold mb-4 text-text-primary text-center">Shopping Cart</h2>
```
Lets not forget our h1 is in Logo.tsx inside Header.tsx!

**Review by maxitect — CHANGES_REQUESTED** [review]
Thanks Marika, this looks really great, shopping cart looks amazing!
Couple of comments to review

File: src/Cart-components/Quantifier.tsx (hunk 2 above)
Comment (MarikaBBB) [reply src/Cart-components/Quantifier.tsx]: Thank for the suggestion. I agreed with you and I created a specific component for these buttons

File: src/pages/ShopCart.tsx [hunk 5]
```diff
@@ -1,95 +1,105 @@
...
+    const updatedCart = cartItems.map((item) =>
+      item.id === VinylId ? { ...item, quantity: item.quantity + 1 } : item
+    );
+    updateCartState(updatedCart);
+  };
+
+  const decreaseQuantity = (VinylId: number) => {
+    const updatedCart = cartItems
+      .map((item) => {
+        if (item.id === VinylId) {
```
Comment (maxitect) [inline src/pages/ShopCart.tsx]:
```suggestion
  const increaseQuantity = (vinylId: number) => {
    const updatedCart = cartItems.map((item) =>
      item.id === vinylId ? { ...item, quantity: item.quantity + 1 } : item
    );
    updateCartState(updatedCart);
  };

  const decreaseQuantity = (vinylId: number) => {
    const updatedCart = cartItems
      .map((item) => {
        if (item.id === vinylId) {
```

File: src/pages/ShopCart.tsx [hunk 6]
```diff
@@ -1,95 +1,105 @@
...
-            )}
+      <div 
+      id="container"
+      className="max-w-6xl mx-auto">
+        {cartItems.length === 0 ? (
+          <p className="text-text-primary text-center">Your cart is empty</p>
+        ) : (
+          cartItems.map((Vinyl) => (
+            <div 
+            id="Vinyl"
```
Comment (maxitect) [inline src/pages/ShopCart.tsx:76]:
```suggestion
            id="vinyl"
```

File: src/pages/ShopCart.tsx [hunk 7]
```diff
@@ -1,95 +1,105 @@
...
+            className="flex items-center justify-between p-4 mb-4 border border-gray-300 rounded-md bg-background-default shadow-sm"
+            key={Vinyl.id}>
+              <img src={Vinyl.image_url} alt={Vinyl.title} 
+              className="w-24 h-24 object-cover rounded-lg"/>
+                <div className="ml-4 flex-grow">
+            <h3 className="text-lg font-medium text-text-primary">{Vinyl.title}</h3>
+            <p className="text-text-secondary">{Vinyl.artist}</p>
+            <p className="font-bold text-dark">£{Vinyl.price.toFixed(2)}</p>
```
Comment (maxitect) [inline src/pages/ShopCart.tsx]:
```suggestion
          cartItems.map((vinyl) => (
            <div 
            id="vinyl"
            className="flex items-center justify-between p-4 mb-4 border border-gray-300 rounded-md bg-background-default shadow-sm"
            key={vinyl.id}>
              <img src={vinyl.image_url} alt={vinyl.title} 
              className="w-24 h-24 object-cover rounded-lg"/>
                <div className="ml-4 flex-grow">
            <h3 className="text-lg font-medium text-text-primary">{vinyl.title}</h3>
            <p className="text-text-secondary">{vinyl.artist}</p>
            <p className="font-bold text-dark">£{vinyl.price.toFixed(2)}</p>
```

File: src/pages/ShopCart.tsx [hunk 8]
```diff
@@ -1,95 +1,105 @@
...
-        </section>
-      );
-    };
- 
-export default ShopCart;
\ No newline at end of file
+              <Quantifier
+                quantity={Vinyl.quantity}
+                onIncrease={() => increaseQuantity(Vinyl.id)}
+                onDecrease={() => decreaseQuantity(Vinyl.id)}
```
Comment (maxitect) [inline src/pages/ShopCart.tsx]:
```suggestion
                quantity={vinyl.quantity}
                onIncrease={() => increaseQuantity(vinyl.id)}
                onDecrease={() => decreaseQuantity(vinyl.id)}
```

**Review by maxitect — APPROVED** [review]
Thank you Marika!
